### PR TITLE
add support for arduino R4

### PR DESCRIPTION
**Pull request by VIPQualityPost**
Add PIO envs and change untested hardware build error to warning.

### CODE REVIEW

File: src/ayab/board.h
Comment (dl1com) [inline src/ayab/board.h:1]: Please add a define guard for the new R4 platform.

**Comment by dl1com** [comment]
As discussed in the last meeting, as we face the need for supporting multiple hardware platforms (UNO, UNO R4, AYAB-ESP32), we'll have to adjust the firmware CI accordingly so it generates and uploads multiple binaries.
So this should be added to this PR already. (building both UNO and R4 binaries).

Plus, the Desktop software shall be capable of handling these different binaries (downloading the binary releases using the manifest file, showing them in the firmware flash UI, be able to flash them to the target (esp32 flash tool required to be shipped)) (I added an issue for this here: https://github.com/AllYarnsAreBeautiful/ayab-desktop/issues/612)

**Comment by dl1com** [comment]
I just had another look at this PR... it seems the base branch is odd, as git tries to rebase 48 commits (which is about the distance to the current state of the main branch.

Where do we want the R4 change to show up? Is it still relevant for 0.95 or is it sufficient to have it in 1.0? (or vice versa?)

**Comment by VIPQualityPost** [comment]
I can fix it so that it's just coming off main.. but this should not be pulled into 1.0 I think, we can roll in support for this alongside ESP32 in a post-v1.0 feature release.

**Pull request closed without merging**